### PR TITLE
variable name fixes and other additions/corrections

### PR DESCRIPTION
**Pull request by megamorf**
* Typo corrections in description and comments
* Variable names changed from camel case to PowerShell best practice
* Added Log function for additional verbose output
* Added missing parameter names in a few places

### CODE REVIEW

**Comment by TravisEz13** [comment]
Please resolve all merge conflicts.  For more information see [GitHowTo - 30. Resolving Conflicts](http://githowto.com/resolving_conflicts )
[Instructions to setup BC4, my favorite merge tool](http://www.scootersoftware.com/support.php?zz=kb_vcs#gitwindows)

**Comment by TravisEz13** [comment]
You should have gotten an email from msftclas no how to sign the cla.

**Comment by TravisEz13** [comment]
It is required that you provide adequate test coverage for the code you change. [Contribution Guidelines -Adding test coverage for DSC resources](https://github.com/PowerShell/DscResources/blob/master/CONTRIBUTING.md#adding-test-coverage-for-dsc-resources)  Please work with us to come up with a plan to cover this code.

Suggestion: Base the tests on the [Unit tests Template](https://github.com/PowerShell/DscResources/blob/master/Tests.Template/unit_template.ps1)

**Comment by megamorf** [comment]
I'll gladly add test coverage for the log function as soon as @NarineM has added his tests to the DSCPullServerSetup/PublishModulesAndMofsToPullServer.psm1 module.

**Comment by megamorf** [comment]
I don't know why the merge conflict happened in the first place but it's resolved now.

File: DSCPullServerSetup/PublishModulesAndMofsToPullServer.psm1 [hunk 1]
```diff
@@ -165,4 +165,4 @@ Function Log
     Write-Verbose "$Date [$($Scope.MyCommand)] :: $Message"
 }
 
-Export-ModuleMember -Function Publish-DSCModuleAndMof
+Export-ModuleMember -Function Publish-DSCModuleAndMof
```
Comment (NarineM) [inline DSCPullServerSetup/PublishModulesAndMofsToPullServer.psm1]: New line is missing.

File: DSCPullServerSetup/PublishModulesAndMofsToPullServer.psm1 [hunk 2]
```diff
@@ -1,147 +1,168 @@
...
+        Log -Scope $MyInvocation -Message "No additional modules are specified to be packaged." 
     }
-    foreach ($module in $listModuleNames)
+    
+    foreach ($Module in $ListModuleNames)
     {
-        $allVersions = Get-Module -Name $module -ListAvailable -Verbose        
-        #package all versions of the module
-        foreach($moduleVersion in $allVersions)
+        $AllVersions = Get-Module -Name $Module -ListAvailable -Verbose
```
Comment (NarineM) [inline DSCPullServerSetup/PublishModulesAndMofsToPullServer.psm1]: Variable names should use camelCase

File: DSCPullServerSetup/PublishModulesAndMofsToPullServer.psm1 [hunk 3]
```diff
@@ -1,147 +1,168 @@
...
-            Compress-Archive -Path "$path\*" -DestinationPath "$source.zip" -Verbose -Force 
-            $version = $moduleVersion.Version.ToString()
-            $newName = "$destination\$name" + "_" + "$version" + ".zip"
+            $Name   = $ModuleVersion.Name
+            $Source = "$Destination\$Name"
+            # Create package zip
+            $Path    = $ModuleVersion.ModuleBase
+            $Version = $ModuleVersion.Version.ToString()
+            Log -Scope $MyInvocation -Message "Zipping $Name ($Version)"
+            Compress-Archive -Path "$Path\*" -DestinationPath "$Source.zip" -Verbose -Force
```
Comment (NarineM) [inline DSCPullServerSetup/PublishModulesAndMofsToPullServer.psm1]: Variable names should use camelCase. Same comment applies to all your changes to this file.

File: DSCPullServerSetup/PublishModulesAndMofsToPullServer.psm1 [hunk 4]
```diff
@@ -1,147 +1,168 @@
...
+    Param(
+        $Date = $(Get-Date),
+        $Scope, 
+        $Message
+    )
+
+    Write-Verbose "$Date [$($Scope.MyCommand)] :: $Message"
+}
+
 Export-ModuleMember -Function Publish-DSCModuleAndMof
```
Comment (NarineM) [inline DSCPullServerSetup/PublishModulesAndMofsToPullServer.psm1]: Should end with new line.

File: DSCPullServerSetup/PublishModulesAndMofsToPullServer.psm1 (hunk 3 above)
Comment (megamorf) [inline DSCPullServerSetup/PublishModulesAndMofsToPullServer.psm1]: Parameters in PowerShell use PascalCase notation.
The documentation only mentions Pascal Case: https://msdn.microsoft.com/en-us/library/dd878270(v=vs.85).aspx#SD02

This is not .NET and parameter names throughout the online documentation and help files use PascalCase. As long as the parameters adhere to these common practices I'd be fine.

Comment (NarineM) [inline DSCPullServerSetup/PublishModulesAndMofsToPullServer.psm1]: For all DSC modules and resources we follow these style guidelines:
https://github.com/PowerShell/DscResources/blob/master/StyleGuidelines.md
Parameter names are pascal case, and variable names are camel case. 
Please revert unnecessary changes for this module.

File: DSCPullServerSetup/PublishModulesAndMofsToPullServer.psm1 [hunk 5]
```diff
@@ -165,4 +165,4 @@ Function Log
     Write-Verbose "$Date [$($Scope.MyCommand)] :: $Message"
 }
 
-Export-ModuleMember -Function Publish-DSCModuleAndMof
```
Comment (NarineM) [inline DSCPullServerSetup/PublishModulesAndMofsToPullServer.psm1]: I do not see any change for this line as compared to original code.

File: DSCPullServerSetup/PublishModulesAndMofsToPullServer.psm1 (hunk 3 above)
Comment (megamorf) [inline DSCPullServerSetup/PublishModulesAndMofsToPullServer.psm1]: This module is neither a dsc resource nor a dsc resource module. It's just coincidentally placed in this github repo because you have write permissions to the repository.

**Comment by megamorf** [comment]
Anyway, I'm not going to bother any further. Please ensure that function parameter names adhere to pascal case notation and refrain from using Write-Host.

File: DSCPullServerSetup/PublishModulesAndMofsToPullServer.psm1 (hunk 3 above)
Comment (NarineM) [inline DSCPullServerSetup/PublishModulesAndMofsToPullServer.psm1]: This module/cmdlet is part of xPSDesiredStateConfiguration module and this is the guidelines we are following: https://github.com/PowerShell/DscResources/blob/master/StyleGuidelines.md

**Comment by megamorf** [comment]
I recommend copying at least the comments - yours were sloppy and inconsistent.

**Comment by NarineM** [comment]
@megamorf ,
other than reverting your changes for the variable names and leaving them to be came case, your other changes apply so please proceed with those.  thanks.

**Comment by NarineM** [comment]
@megamorf ,
 thanks for your contribution.
